usr/share/rear/build/default/501_check_ssh_keys.sh

```diff
@@ -0,0 +1,35 @@
+# remove unprotected SSH keys unless configured to keep them
+is_true "$SSH_UNPROTECTED_PRIVATE_KEYS" && return
+
+# caveat: this code will only detect SSH key files for root, not for other users
+key_files=(
+    $ROOTFS_DIR/root/.ssh/{identity,id_dsa,id_ecdsa,id_ed25519,id_rsa}
```

Comment:
What if I have key _~/.ssh/master_key_to_all_servers_passphrase_unprotected_please_dont_steal_me_ ?
If unprotected keys should be excluded, then I'd expect that ALL unprotected keys are excluded not only those which are hard coded ...

Reply:
The main problem is that so far we haven't yet figured out how to reliably detect private SSH keys.

My code will check your magic master key **if you refer** to it in the SSH client configuration. For example with this config line it will be "caught":
```
Host *
   IdentitiyFile ~/.ssh/master_key_to_all_servers_passphrase_unprotected_please_dont_steal_me
```

If you have a key that you don't refer in the config at all then it will indeed be copied to the rescue media.

Reply:
Looks like a perfect example for the documentation.

I think ReaR cannot be expected to be acually intelligent.
As long as ReaR tries to some reasonable things to keep
its recovery system by default free of secrets it is o.k. for me.
If the admin has some special setup anything might leak
into the recovery system and that should be documented
because with "KEEP_BUILD_DIR=1" the admin can really
easily check what there actually is in his recovery system.

Reply:
@schlomo just from top of my head:
```
# cd ~/.ssh
# file * | grep "private key"
key2:                PEM RSA private key
key1:             PEM RSA private key
```

@jsmeix 
> I think ReaR cannot be expected to be acually intelligent.

Fully agree, and that is the reason why we should not even try to fix things like this. If admin wants to be insecure I don't see a reason why we should enforce him something different.

"UNIX was not designed to stop its users from doing stupid things, as that would also stop them from doing clever things." (Doug Gwyn).

Reply:
@schlomo `magic` does not recognize private key on all OSes so forget about it.
Without reliable detection of all private keys in ~./ssh dir this PR does not solve a thing ...

Reply:
@gozora I had similar thoughts. `$ grep -l BEGIN.*PRIVATE.*KEY *` seems to catch all the keys, both in the old and new format. So yes, we could scan the `.ssh` directory, but we would fail to catch keys in other directories.

I actually don't think that this PR has no value. It does increase the default security for the "just user" without increasing the run time too much. OTOH I would say that scanning all files in the rescue media might be already too time consuming. Or maybe the run time is actually totally irrelevant for our users and just annoys us developers.

Reply:
@schlomo ssh key detection is available in file-5.3.2. This is how guys from `file` do the [detection](https://github.com/file/file/blob/master/magic/Magdir/ssl).

Maybe I was a bit hard with my judgement before. Maybe you could rename this PR to something like:
"Avoid leaking unprotected identity, id_dsa, id_ecdsa, id_ed25519, id_rsa files for root home directory into rescue medium"

V.

Reply:
What about sticking to the _secure by default_ principle and copying only white-listed files from `/root/.ssh` and `/etc/ssh`? Then, if the user insists, let him add additional files on request. The latter would require one configuration change now, but not restrict any use cases beyond what is possible currently.